libphal/phal_sbe.C

```diff
@@ -374,6 +385,80 @@ sbeError_t captureFFDC(struct pdbg_target *target, bool coSuccess)
 	}
 }
 
+sbeError_t capturePfutFFDC(struct pdbg_target *target, std::vector<FFDCEntry>& lientry)
+{
+	pdbg_target* opTarget = getPibTarget(target);
+	bufPtr_t bufPtr;
+	uint32_t ffdcLen = 0;
+	uint32_t status = 0;
+
+	if (sbe_ffdc_get(opTarget, &status, bufPtr.getPtr(), &ffdcLen)) {
+		log(level::ERROR, "Failed to get SBE FFDC data (%s)",
+		pdbg_target_path(opTarget));
+		return sbeError_t(exception::SBE_FFDC_GET_FAILED);
+	}
+
+	if (status == (SBEFIFO_PRI_UNKNOWN_ERROR | SBEFIFO_SEC_HW_TIMEOUT)) {
+		return sbeError_t(exception::SBE_CMD_TIMEOUT);
+	}
+
+	if (!ffdcLen) {
+		return sbeError_t(exception::SBE_FFDC_NO_DATA);
+	}
+	uint32_t offset = 0;
+	while (offset < ffdcLen) {
+		uint16_t magic_bytes =
+			ntohs(*reinterpret_cast<uint16_t *>(bufPtr.getData() + offset));
+		FFDCEntry entry;
+		uint16_t lenInWords = 0;
+		if (magic_bytes == ffdcMagicCode) {
```

Comment:
Suggestion: Would you consider using a template in the below scenario? The code is repeated. Please feel free to ignore, if not applicable.

template <typename HeaderType>
void processFfdcEntry(const uint8_t* base, size_t offset, uint16_t magic_bytes, Entry& entry)
{
    const HeaderType* hdr = reinterpret_cast<const HeaderType*>(base + offset);
    uint16_t lenInWords = ntohs(hdr->lengthinWords);
...
...
}

const uint8_t* base = bufPtr.getData();

if (magic_bytes == ffdcMagicCode)
{
    processFfdcEntry<p10FfdcHeader>(base, offset, magic_bytes, entry);
}
else if (magic_bytes == pozFfdcMagicCode)
{
    processFfdcEntry<pozFfdcHeader>(base, offset, magic_bytes, entry);
}

Reply:
fixed this issue, cannot use template as I do not want the caller to pass any extra arguments to used helper methods.